### PR TITLE
feat(pipeline): pipeline add canceling status

### PR DESCRIPTION
**Pull request by sfwn**
#### What this PR does / why we need it:

pipeline add canceling status

#### Which issue(s) this PR fixes:

- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=343878&iterationID=1440&type=TASK)

#### Specified Reviewers:

/assign @chengjoey 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    pipeline add canceling status          |
| 🇨🇳 中文    |   pipeline 增加 "取消中" 状态           |

### CODE REVIEW

File: internal/apps/dop/component-protocol/components/auto-test-scenes/rightPage/fileDetail/fileExecute/executeHead/executeHistory/executeHistoryPop/executeHistoryTable/render.go [hunk 1]
```diff
@@ -228,7 +228,7 @@ func (e *ExecuteHistoryTable) getStatus(req apistructs.PipelineStatus) map[strin
 		res["status"] = "error"
 	} else if req.IsSuccessStatus() {
 		res["status"] = "success"
-	} else if req.IsReconcilerRunningStatus() {
+	} else if req.IsRunningStatus() || req.IsCancelingStatus() {
```
Comment (chengjoey) [inline internal/apps/dop/component-protocol/components/auto-test-scenes/rightPage/fileDetail/fileExecute/executeHead/executeHistory/executeHistoryPop/executeHistoryTable/render.go:231]: use `IsReconcilerRunningStatus` func

File: internal/core/openapi/legacy/component-protocol/scenarios/auto-test-plan-detail/components/executeHistoryTable/render.go [hunk 2]
```diff
@@ -241,7 +241,7 @@ func getStatus(req apistructs.PipelineStatus, i18nLocale *i18n.LocaleResource) m
 		res["status"] = "error"
 	} else if req.IsSuccessStatus() {
 		res["status"] = "success"
-	} else if req.IsReconcilerRunningStatus() {
+	} else if req.IsReconcilerRunningStatus() || req.IsCancelingStatus() {
```
Comment (chengjoey) [inline internal/core/openapi/legacy/component-protocol/scenarios/auto-test-plan-detail/components/executeHistoryTable/render.go:244]: `IsReconcilerRunningStatus` is enough

File: internal/apps/dop/component-protocol/components/auto-test-scenes/rightPage/fileDetail/fileExecute/executeHead/executeHistory/executeHistoryPop/executeHistoryTable/render.go (hunk 1 above)
Comment (sfwn) [reply internal/apps/dop/component-protocol/components/auto-test-scenes/rightPage/fileDetail/fileExecute/executeHead/executeHistory/executeHistoryPop/executeHistoryTable/render.go]: adjusted

File: internal/core/openapi/legacy/component-protocol/scenarios/auto-test-plan-detail/components/executeHistoryTable/render.go (hunk 2 above)
Comment (sfwn) [reply internal/core/openapi/legacy/component-protocol/scenarios/auto-test-plan-detail/components/executeHistoryTable/render.go]: adjusted

**Review by chengjoey — APPROVED** [review]

**Pull request merged**